qcodes/instrument_drivers/Keysight/keysight_e4980a.py

```diff
@@ -0,0 +1,313 @@
+            docstring="Enables or disable SHORT correction."
+        )
+
+
+class KeysightE4980A(VisaInstrument):
+    """
+    QCodes driver for E4980A Precision LCR Meter
+    """
+    def __init__(self, name, address, terminator='\n', **kwargs):
+        """
+        Create an instance of the instrument.
+
+        Args:
+            name: Name of the instrument instance
+            address: Visa-resolvable instrument address.
+        """
+        super().__init__(name, address, terminator=terminator, **kwargs)
+
+        self._measurement_pair = MeasurementPair(
+            "CPD",
+            ("capacitance", "dissipation_factor"),
+            ("F", "")
+        )
+
+        self.add_parameter(
+            "frequency",
+            get_cmd=":FREQuency?",
+            set_cmd=":FREQuency {}",
+            get_parser=float,
+            unit="Hz",
+            vals=Numbers(20, 2E6),
+            docstring="Gets and sets the frequency for normal measurement."
+        )
+
+        self.add_parameter(
+            "current_level",
+            get_cmd=":CURRent:LEVel?",
+            set_cmd=":CURRent:LEVel {}",
+            get_parser=float,
+            unit="A",
+            vals=Numbers(0, 0.1),
+            docstring="Gets and sets the current level for measurement signal."
+        )
+
+        self.add_parameter(
+            "voltage_level",
+            get_cmd=":VOLTage:LEVel?",
+            set_cmd=":VOLTage:LEVel {}",
+            get_parser=float,
+            unit="V",
+            vals=Numbers(0, 20),
+            docstring="Gets and sets the voltage level for measurement signal."
+        )
+
+        self.add_parameter(
+            "measurement_function",
+            get_cmd=":FUNCtion:IMPedance?",
+            set_cmd=self._set_measurement
+        )
+
+        self.add_parameter(
+            "range",
+            get_cmd=":FUNCtion:IMPedance:RANGe?",
+            set_cmd=":FUNCtion:IMPedance:RANGe {}",
+            unit='Ohm',
+            vals=Enum(0.1, 1, 10, 100, 300, 1000, 3000, 10000, 30000, 100000),
+            docstring="Selects the impedance measurement range, also turns "
+                      "the auto range function OFF."
+        )
+
+        self.add_parameter(
+            "system_errors",
+            get_cmd=":SYSTem:ERRor?",
+            docstring="Returns the oldest unread error message from the event "
+                      "log and removes it from the log."
+        )
+
+        self.add_submodule(
+            "_correction",
+            Correction4980A(self, "correction")
+        )
+
+        self.connect_message()
+
+    @property
+    def correction(self):
+        return self.submodules['_correction']
+
+    @property
+    def measure_impedance(self) -> MeasurementPair:
+        return self._get_complex_impedance()
+
+    @property
+    def measurement(self) -> MeasurementPair:
+        return self._measurement()
+
+    def _get_complex_impedance(self) -> MeasurementPair:
```

Comment:
what is the difference between using `measure_impedance` and `measurement` with RX? specifically, is there a reason for them having slightly different instrument commands? if not, then why not let `measure_impedance` resuse `measurement` in its implementation?

Reply:
They are doing the same thing. The reason I have the stand alone impedance measurement here is to provide a shortcut without change the "main" measurement function. I guess this is why the instrument have both at the same time.